src/nitpackage.nit

```diff
@@ -287,6 +393,109 @@ redef class ConfigTree
 	end
 end
 
+# A Makefile for the Nit project
+class NitMakefile
+
+	# Nit files to compile
+	var nit_files: Array[String]
+
+	# List of rules to add in the Makefile
+	fun rules: Array[MakeRule] do
+		var rules = new Array[MakeRule]
+
+		var rule_all = new MakeRule("all")
+		if nit_files.not_empty then
+			rules.add rule_all
```

Comment:
I think we basically always need an `all` target since `make` without parameter will make the first target of the file. An `all` rule, possibly empty, will be this first target instead of whatever comes next in the Makefile